bot/src/com/mygdx/game/bot/screen/GameScreen.java

```diff
@@ -132,17 +136,21 @@ private void runBotIteration() {
       return;
     }
     log.info("start turn");
-    var unit = nextUnitUtil.selectNextUnit();
-    if (unit == 0xC0FFEE) { // todo ensure the case where a unit that has moveRange and attack but no possible
+    IntArray availableUnits = unitUtil.getAllUnits();
+    if (availableUnits == null) { // todo ensure the case where a unit that has moveRange and attack but no possible
       // ways to use it be skipped
       log.info("end turn");
       endTurn();
       return;
     }
-    var field = nextFieldUtil.selectFieldInRangeOfUnit(unit);
-    log.info("moving entity");
-    moveEntityService.moveEntity(unit, coordinatesComponentMapper.get(field));
-    botAttackUtil.attack(unit);
+    for (int unit:availableUnits.items) {
+      botAttackUtil.attack(unit);
+      botMoveUtil.move(unit);
```

Comment:
O tym gadaliśmy już IIRC, że jak bot robi modyfikację stanu gry to musi czekać na odpowiedź serwera, a nie robić losową rzecz

Reply:
a, ale to było dodane wcześniej widzę